src/OpenApiType.php

```diff
@@ -225,7 +225,13 @@ public static function resolve(string $context, array $definitions, ParamTagValu
 		}
 
 		if ($node instanceof GenericTypeNode && $node->type->name === 'array' && count($node->genericTypes) === 2 && $node->genericTypes[0] instanceof IdentifierTypeNode) {
-			if ($node->genericTypes[0]->name === 'string') {
+			if (in_array($node->genericTypes[0]->name, ['string', 'lowercase-string', 'non-empty-string', 'non-empty-lowercase-string'], true)) {
+				if (str_starts_with($node->genericTypes[0]->name, 'non-empty-')) {
+					// FIXME Make sure the keys are non-empty
+				}
+				if (str_ends_with($node->genericTypes[0]->name, 'lowercase-string')) {
+					// FIXME Make sure the keys are lowercase only
+				}
```

Comment:
```suggestion
```
Psalm already does this part for us, it will fail if the key type is specified as non-empty or lowercase and it actually isn't.

Reply:
but it doesn't show in the OpenAPI specs. Wouldn't it be helpful for mobile devs?

Reply:
Yes, but lowercase can not be documented via JSON schema in general. Non-empty can be documented in general (and it is already), but in this particular case it doesn't work because the key type is not explicitly documented via JSON schema.
The only help we could give is adding a comment.